File: 1894-merge-strings-alternately/1894-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string merged = "";
        int i = 0, j = 0;
        
        while(i < word1.size() && j < word2.size()){
            merged += word1[i++];
            merged += word2[j++];
        }

        merged += word1.substr(i) + word2.substr(j);

        return merged;
    }
};

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        length = len(max(word1, word2, key=len))
        word = ""

        for i in range(length):
            if(i < len(word1)):
                word += word1[i]
            
            if(i < len(word2)):
                word += word2[i]

        return word